lib/mongo/socket/unix.rb

```diff
@@ -56,7 +53,7 @@ def connect!
       # @since 2.0.0
       def initialize(path, timeout, family)
         @path, @timeout = path, timeout
-        super(family)
+        super(family, path)
       end
```

Comment:
Thanks so much for looking into this.

Your changes look good. My own suggestion is that we make this method, Socket::Unix#initialize look like this

```ruby
def initialize(path, timeout, family)
  @path, @timeout = path, timeout
  @socket = ::Socket.unix(path)
  set_socket_options(@socket)
end
```

instead of calling super with an extra argument. That means the changes in socket.rb wouldn't be necessary. What do you think?